src/components/ebay-carousel/index.js

```diff
@@ -85,120 +85,104 @@ function init() {
         this.simulateDotClick(this.state.slide);
-    } else if (this.state.isContinuous) {
-        this.performSlide();
     }
+    this.processMovement();
 }
 
-function update_index() { // eslint-disable-line camelcase
-    this.performSlide();
-}
-
-function handleNext(originalEvent) {
-    if (!this.state.nextControlDisabled) {
-        if (this.state.isDiscrete) {
-            this.simulateDotClick(this.state.slide + 1);
-        } else if (this.state.isContinuous) {
-            this.setState('index', this.calculateNextIndex());
-        }
-        emitAndFire(this, 'carousel-next', { originalEvent });
+/**
+ * High level function called for movement upon changing index (via UI and API)
+ * Exits early if there was no actual change to index
+ */
+function processIndexChange() {
+    // TODO: API manipulation is disabled in native scroll case
+    if (this.usesNativeScroll) {
+        return;
     }
-}
 
-function handlePrev(originalEvent) {
-    if (!this.state.prevControlDisabled) {
-        if (this.state.isDiscrete) {
-            this.simulateDotClick(this.state.slide - 1);
-        } else if (this.state.isContinuous) {
-            this.setState('index', this.calculatePrevIndex());
-        }
-        emitAndFire(this, 'carousel-prev', { originalEvent });
+    if (this.state.index > this.lastIndex) {
+        this.setState('index', this.lastIndex);
     }
-}
 
-// TODO: add carousel-dot event and remove simulated dot clicks
-function handleDotClick(e) {
-    const newSlide = parseInt(e.target.getAttribute('data-slide'));
-    emitAndFire(this, 'carousel-slide', { slide: newSlide });
-    this.setState('slide', newSlide);
-    this.setState('index', (this.state.itemsPerSlide * (newSlide - 1)));
-    this.update_index(); // FIXME: why isn't this called from this.setState('index')?
-}
+    if (this.containerWidth >= this.allItemsWidth || this.state.index < 0) {
+        this.setState('index', 0);
+    }
 
-function simulateDotClick(slide) {
-    if (slide >= 1 && slide <= this.state.totalSlides) {
-        this.el.querySelector(`[data-slide="${slide}"]`).click();
+    if (this.state.index === this.activeIndex) {
+        return;
     }
-}
 
-function updateDots() {
-    this.setState('activeDot', (this.state.lastVisibleIndex + 1) / this.state.itemsPerSlide);
+    this.activeIndex = this.state.index;
+
+    this.processMovement();
 }
 
 /**
- * High level slide called for initialization and data change to index (via UI and API)
- * @param {Integer} index
+ * Handle movement to current state.index
  */
-function performSlide() {
-    // FIXME: API manipulation is disabled in native scroll case
-    if (this.state.index >= 0 && this.state.index <= this.lastIndex && !this.usesNativeScroll) {
-        const oldFirstVisibleIndex = this.state.firstVisibleIndex;
-        const oldLastVisibleIndex = this.state.lastVisibleIndex;
+function processMovement() {
+    this.processMovementFrame = requestAnimationFrame(() => {
```

Comment:
We should probably `cancelAnimationFrame` before this line. Just incase they change the state twice in a frame we can avoid doing extra work.